File: src/main/java/com/sih/crowdsource/dto/request/ComplaintRequest.java
package com.sih.crowdsource.dto.request;

import com.sih.crowdsource.enums.ComplaintPriority;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ComplaintRequest {

    private String title;
    private String description;
    private String location;
    private ComplaintPriority priority;
    private Long departmentId;
}

File: Flutter_FrontEnd/crowd_source_civic_app/lib/Data/models/complaint_request.dart
import 'complaint_priority.dart';

class ComplaintRequest {
  final String title;
  final String description;
  final String location;
  final ComplaintPriority priority;
  final int departmentId;

  ComplaintRequest({
    required this.title,
    required this.description,
    required this.location,
    required this.priority,
    required this.departmentId,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'location': location,
      'priority': priority.name,
      'departmentId': departmentId,
    };
  }
}
